crates/api/src/local_user/donation_dialog_shown.rs

```diff
@@ -1,16 +1,22 @@
 use actix_web::web::{Data, Json};
-use chrono::Utc;
-use lemmy_api_common::{context::LemmyContext, SuccessResponse};
+use chrono::{DateTime, Utc};
+use lemmy_api_common::{context::LemmyContext, person::DonationDialogShown, SuccessResponse};
 use lemmy_db_schema::source::local_user::{LocalUser, LocalUserUpdateForm};
 use lemmy_db_views::structs::LocalUserView;
 use lemmy_utils::error::LemmyResult;
 
 pub async fn donation_dialog_shown(
   context: Data<LemmyContext>,
   local_user_view: LocalUserView,
+  data: Json<DonationDialogShown>,
 ) -> LemmyResult<Json<SuccessResponse>> {
+  let last = if data.hide_permanently.unwrap_or_default() {
+    DateTime::<Utc>::MAX_UTC
+  } else {
+    Utc::now()
+  };
```

Comment:
Instead of doing this as a param, add this as a column to `local_user`, and allow them to save the setting. 

Then you can get this field from the `local_user_view`

That way it'll apply across all apps, and front ends won't have to remember this.

Reply:
Hide permanently just sets `last_donation_notification` in the future, so the dialog will never be shown. This way is much easier all around:
- UIs only need a single check `last_donation_notification < now() - '1 year'`, nothing else
- Doesnt require separate db column
- The dialog can be shown with buttons "Donate / Hide / Hide Permanently" so its all in one place. Otherwise the hide option would be hard for users to find.

Reply:
Alright I suppose that works. Forgot that `last_donation_notification` is already saving that on `local_user`